contrib/babelfishpg_tsql/src/iterative_exec.c

```diff
@@ -99,44 +99,75 @@ static int exec_stmt_raiserror(PLtsql_execstate *estate, PLtsql_stmt_raiserror *
 	int		severity = -1;
 	int		state = -1;
 
-	/* Read parameters of RAISERROR statements */
-	read_raiserror_params(estate, stmt->params, stmt->paramno, &msg, &msg_id, &severity, &state);
-	msg = pstrdup(msg);
+	PLtsql_expr *temp_expr;
+	int stmt_len = -1;
+	char *param_text = NULL;
+	char *query_text = NULL;
+	char *explain_text = NULL;
+
+	if (pltsql_explain_only) {
+		/* calculate stmt length */
+		stmt_len = 0;
+		for (int i = 0; i < stmt->paramno; i++)
+		{
+			temp_expr = (PLtsql_expr *) list_nth(stmt->params, i);
+			/* minus 6 is stripping 'SELECT ' and adding one space for a comma */
+			stmt_len += strlen(temp_expr->query) - 6;
+		}
+		char *query_text = (char *) palloc0(sizeof(char) * stmt_len);
```

Comment:
Instead of using `palloc` and `strncat`, let's try to use `StringInfoData` if possible.